### ISSUE TITLE
[FEATURE/BUG] Define guidelines for readers execution in the terminal 

### ISSUE THREAD

**Issue by ghiggi** [ltelab/disdrodb#79]
The current only way to execute a reader is to clone the disdrodb repo, install disdrodb in dev-mode (see https://disdrodb-dev.readthedocs.io/en/latest/install.html#installation-for-contributors) and then call the reader as follow: 

`python python_file_path raw_dir  processed_dir [parameters]`

**Currently, the disdrodb package installed with pip install disdrodb does not contain any reader !**
**--> This will hopefully be fixed soon ... <---** @regislon 

But also if the installed disdrodb package would now include all the implemented readers present in the L0 directory, 
the user should: 

1. Identify the location of the compiled package and the directory where the readers are located. 
   If the disdrodb installation is done within a conda environment, the path would be 
`/home/ghiggi/anaconda3/envs/disdro_install/lib/python3.9/site-packages/disdrodb/L0/readers/<data_source>`.
This type of directory path is awful and far from straightforward to identify for a beginner user  
2. Enter the identified directory and call `python <reader_file_name>.py  <raw_dir>  <processed_dir> [parameters]`

We therefore need to implement a clear way to run the readers ! 

#--------------------------------------------------------------------------------------------------------------------------------------------

Here below I list the 2 possible options that I think could be implemented to solve the current problems: 

1. All readers within `disdrodb/L0/readers` become listed as `scripts` in the `setup.py` file.  [See here the scripts doc](https://docs.python.org/3/distutils/setupscript.html#installing-scripts).

   Therefore a reader within `disdrodb/L0/readers`  called `reader_<campaign_name>` could be simply run from the terminal by typing 
   `reader_<campaign_name> raw_dir  processed_dir [parameters]`

   Following this approach, we should ensure:  
   - a strict naming convention for the readers 
   - unique naming of readers. However note that some `campaign_name` are present in multiple `data_source` (when multiple data_sources participated to a given campaign), so the readers name are not unique now.
   
   I think this option is not very sustainable also because of: 
   - the long list of readers to be registered as scripts;
   - the need to modify the `setup.py` every time a new reader is developed.

2.  As an alternative, we can define a single python script (e.g. `DISDRODB_L0_reader.py`) that can be run from the terminal with:
      `DISDRODB_L0_reader <data_source> <campaign_name> <raw_dir> <processed_dir> [parameters]`
      
     To do this, we need to implement a `get_reader(data_source, campaign_name)` function that searches the available readers (#36) and/or imports the requested L0 reader based on the `data_source` and `campaign_name` . 
 
    Then the `DISDRODB_L0_reader.py` general script file can be defined as follow,  

```
import click
from disdrodb.L0 import click_L0_readers_options, get_reader 
 
# -------------------------------------------------------------------------.
# CLIck Command Line Interface decorator
@click.command()  # options_metavar='<options>'
@click.argument('data_source', type=click.Path(exists=True), metavar='<data_source>')
@click.argument('reader_name', metavar='<reader_name>') # or campaign_name TBD
@click_L0_readers_options
def main(
         data_source,
         reader_name,
         # Reader options 
         raw_dir,               # or base_directory 
         processed_dir,    # or base_directory
         l0a_processing=True,
         l0b_processing=True,
         keep_l0a=False,
         force=False,
         verbose=False,
         debugging_mode=False,
         lazy=True,
         single_netcdf=True, 
         ):

     reader = get_reader(data_source, reader_name) 
     reader(raw_dir=raw_dir,
         processed_dir=processed_dir, 
         l0a_processing=l0a_processing,
         l0b_processing=l0b_processing,
         keep_l0a=keep_l0a,
         force=force,
         verbose=verbose,
         debugging_mode=debugging_mode,
         lazy=lazy,
         single_netcdf=single_netcdf)

if __main__:
        main() 

```
If we follow this second option, we need to modify the L0 readers to: 
    -  remove the `if __main__:  reader()` piece of code from the bottom of each L0 reader file.  
    -  remove the click wrapper from each L0 reader file.
 
Following this option 2 will not enable to run anymore in dev-mode the individual L0 readers files from the terminal.
However this option 2 would also enable to run the process of the datasets/campaign within python for loops: 
  ```
  # ... 
  data_source = "EPFL"
  raw_data_source_dir = os.path.join(raw_dir, data_source)
  processed_data_source_dir = os.path.join(processed_dir, data_source)
  list_campaigns = get_available_readers(data_source=data_source)
  # Loop and process all campaigns of a specific data_source
  for campaign_name in list_campaigns: 
      reader = get_reader(data_source, campaign_name)
      reader(os.path.join(raw_data_source_dir, campaign_name), 
                 os.path.join(processed_data_source_dir, campaign_name), 
                 **kwargs)
 ```

Additionally, by adopting option 2, we could simply import the readers by doing e.g. `from disdrodb.L0.readers.EPFL.reader_LOCARNO_2018 import reader`. By changing the naming pattern of the readers to some pattern like `<campaign_name>.py` then the import would become something like  `from disdrodb.L0.readers.EPFL.LOCARNO_2018 import reader`. 

In conclusion, I think the @charlottegiseleweil suggestion in #52  of using `base_directory`, `data_source` and `campaign_name`  could be simply imposed in the `DISDRODB_L0_reader.py`, while maybe leaving `raw_dir` and `processed_dir` as arguments in the L0 readers files to provide greater flexibility to the disdrodb user ...

**Comment by regislon** [comment]
Hi @ghiggi 

If I understand you well. There will be two means of running the reader : 

1.  by command line

e.g,: 
`python disdrodb\L0\run_reader_cmd.py NETHERLANDS DELFT "C:\projects\disdrodb_data\data\DISDRODB\Raw\NETHERLANDS\DELFT" "C:\projects\disdrodb_data\data\DISDRODB\Processed\NETHERLANDS\DELFT" `

... and others parameters if not default

2. by calling the funtion 

``` python
from disdrodb.L0.readers.NETHERLANDS.DELFT import reader

raw_dir = "C:\\projects\\disdrodb_data\\data\\DISDRODB\\Raw\\NETHERLANDS\\DELFT"
processed_dir = "C:\\projects\\disdrodb_data\\data\\DISDRODB\\Processed\\NETHERLANDS\\DELFT"
l0a_processing=True
l0b_processing=False
keep_l0a=True
force=True
verbose=True
debugging_mode=True
lazy=True
single_netcdf=True

reader(raw_dir,processed_dir,l0a_processing,l0b_processing,keep_l0a,force,verbose,debugging_mode,lazy,single_netcdf)
```

Am I on the right track ?  both examples currently run, modifications into this [commit](https://github.com/EPFL-ENAC/LTE-disdrodb/commit/a25d7979092bba7751f2fe4c3618aba37efaa19a) (on our fork, but you may have access). 

I haven't done a pull request yet, still a work in progess, just want to know if I'm not à côté de la plaque.

Any comments / remarks are welcome @ghiggi

**Comment by ghiggi** [comment]
Yes, it's exactly like that. 
Note that instead of running `python disdrodb\L0\run_reader_cmd.py` it will be possible to simply run `run_reader_cmd ... `  if we set up stuff correctly in `setup.py` and in the `run_reader_cmd.py` file ;) 
Maybe we could rename `run_reader_cmd.py` as `run_DISDRODB_L0_reader.py` or `run_disdrodb_l0_reader.py`
